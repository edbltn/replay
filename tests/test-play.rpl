strat {A, B};

Game g = Game[2 | {(A,A) -> (1, 1)}];

Strategy s = Strategy[2|2|{state0: A, (A,B) -> state1
                                    | (A,A) -> state1;
                           state1: B, (A,B) -> state0
                                    | (B,B) -> state0;}];

Player p1 = Player[s|0.5];
Player p2 = Player[s|1.0];

println(p1.payoff);
println(p1.state);
println(p1.rounds);

for i in [1:5] {
  p1, p2 % g;
}

println(p1.strategy);
println(p1.payoff);
println(p1.state);
println(p1.rounds);
