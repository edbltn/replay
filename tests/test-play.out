0
0
0
State 0: play 0
( 0 0 ) -> state 1
( 0 1 ) -> state 1
( 1 0 ) -> state 0
( 1 1 ) -> state 0
State 1: play 1
( 0 0 ) -> state 1
( 0 1 ) -> state 0
( 1 0 ) -> state 1
( 1 1 ) -> state 0

1.3125
1
5
